apps/emqx/test/emqx_common_test_helpers.erl

```diff
@@ -950,3 +964,24 @@ latency_up_proxy(off, Name, ProxyHost, ProxyPort) ->
         [],
         [{body_format, binary}]
     ).
+
+%%-------------------------------------------------------------------------------
+%% Testcase teardown utilities
+%%-------------------------------------------------------------------------------
+
+get_on_exit_callbacks(Id) ->
+    persistent_term:get({?MODULE, on_exit, Id}, []).
+
+put_on_exit_callbacks(Id, Funs) ->
+    persistent_term:put({?MODULE, on_exit, Id}, Funs).
+
+on_exit(Id, Fun) ->
+    Callbacks = get_on_exit_callbacks(Id),
+    put_on_exit_callbacks(Id, [Fun | Callbacks]).
+
+%% should be called at `end_per_testcase'.
```

Comment:
If we use `on_exit` per testcase, couldn't we just use the process dict (then we won't need Id) or a linked process (then we won't need call run_on_exit_callbacks either)?

Reply:
I think it looks nicer now.  Thank you!  :beers: